include/aspect/postprocess/boundary_velocity_residual_statistics.h

```diff
@@ -0,0 +1,128 @@
+/*
+  Copyright (C) 2014 - 2019 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+
+#ifndef _aspect_postprocess_boundary_velocity_residual_statistics_h
+#define _aspect_postprocess_boundary_velocity_residual_statistics_h
+
+#include <aspect/postprocess/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/boundary_velocity/gplates.h>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+
+    /**
+     * A postprocessor that computes some statistics about the velocity at
+     * different parts of the boundary.
+     *
+     * @ingroup Postprocessing
+     */
+    template <int dim>
+    class BoundaryVelocityResidualStatistics : public Interface<dim>, public ::aspect::SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Evaluate the solution for some velocity boundary statistics.
+         */
+        std::pair<std::string,std::string>
+        execute (TableHandler &statistics) override;
+
+        /**
+          * This function reads the specified input velocity data files, i.e., either an ascii data file or
+          * a file from the GPlates model.
+          */
+        void initialize () override;
+
+        /**
+         * Evaluate the boundary velocity residual solution at a point.
```

Comment:
I think this explanation needs to be a bit more clear. The function only gets a position, therefore it can not compute a residual (or the solution), all it can do is evaluate the boundary velocity data that is read from the data files.